core/unit_test/TestIsPointer.cpp

```diff
@@ -0,0 +1,28 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <Kokkos_Core.hpp>
+
+// compile-only test
+void test_is_pointer() {
+  using namespace Kokkos::Impl;
+
+  static_assert(is_pointer_v<double *>, "");
```

Comment:
```suggestion
  static_assert(is_pointer_v<double *>);
```
drop all these empty trailing string literal error messages.